src/main/java/io/orten/nano/util/Database.java

```diff
@@ -41,7 +42,9 @@ public static boolean saveOrganization(Organization org) {
         try (Session s = sessionFactory.openSession()) {
             s.beginTransaction();
             Long orgId = org.getOrganizationId();
-            if (getOrganization(orgId) == null) {
+            String orgNo= org.getOrganizationNumber();
+            boolean flag = Pattern.matches("\\w{6}-\\w{4}",orgNo);
+            if ((getOrganization(orgId) == null) && (flag==true)) {
```

Comment:
flag==true is redundant.
just flag should be enough.
(you are checking if true is equal to true, which returns true. Or if false is equal to true, which returns false.)